lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -66,6 +66,11 @@ translate({'__block__', Meta, Args}, S) when is_list(Args) ->
 translate({'__CALLER__', Meta, Atom}, S) when is_atom(Atom) ->
   {{var, ?ann(Meta), '__CALLER__'}, S#elixir_erl{caller=true}};
 
+translate({'__STACKTRACE__', Meta, Atom}, S = #elixir_erl{stacktrace={Var, _}}) when is_atom(Atom) ->
+  {{var, ?ann(Meta), Var}, S#elixir_erl{stacktrace={Var, true}}};
+translate({'__STACKTRACE__', Meta, Atom}, S) when is_atom(Atom) ->
+  elixir_errors:form_error(Meta, S#elixir_erl.file, ?MODULE, invalid_stacktrace);
```

Comment:
We cannot raise errors inside elixir_erl pass. All validation should happen during expansion. We even removed the `.file` field in master. :)